django/utils/http.py

```diff
@@ -303,3 +306,62 @@ def is_safe_url(url, host=None):
         return False
     return ((not url_info.netloc or url_info.netloc == host) and
             (not url_info.scheme or url_info.scheme in ['http', 'https']))
+
+
+# Function lifted from its respective module in order to implement
+# several checks which were not possible with the stdlib, namely limiting
+# the number of fields that will be parsed.
+#
+# Copyright (C) 2013 Python Software Foundation, see license.python.txt for
+# details.
+def limited_parse_qsl(qs, keep_blank_values=False,
+               encoding='utf-8', errors='replace', fields_limit=None):
+    """
+    Parse a query given as a string argument and return a list.
+
+    Lifted from urlparse with an additional "fields_limit" argument.
+
+    Arguments:
+
+    qs: percent-encoded query string to be parsed
+
+    keep_blank_values: flag indicating whether blank values in
+        percent-encoded queries should be treated as blank strings.  A
+        true value indicates that blanks should be retained as blank
+        strings.  The default false value indicates that blank values
+        are to be ignored and treated as if they were  not included.
+
+    encoding and errors: specify how to decode percent-encoded sequences
+        into Unicode characters, as accepted by the bytes.decode() method.
+
+    fields_limit: maximum number of fields parsed or an exception
+        is raised. None means no limit and is the default.
+    """
+    if fields_limit:
+        pairs = FIELDS_MATCH.split(qs, fields_limit)
+        if len(pairs) > fields_limit:
+            raise TooManyFieldsSent('Too many fields sent. Check DATA_UPLOAD_MAX_NUMBER_FIELDS.')
+    else:
+        pairs = FIELDS_MATCH.split(qs)
+    r = []
+    for name_value in pairs:
+        if not name_value:
+            continue
+        nv = name_value.split('=' if six.PY3 else b'=', 1)
```

Comment:
Yes, the original line is misleading. It should just read
```python
    nv = name_value.split('=')
```
because when `not six.PY3` (that is, in your `else` part), `b'='` and `'='` are two spellings of the same entity.